File: algorithms/implementation/the-birthday-bar/the-birthday-bar-python/tests/test_birthday.py
from birthday import birthday

class testebirthday:
    
    def testeCaso1(self):
        
        n = 5
        s = [1, 2, 1, 3, 2]
        d = 3
        m = 2
        
        esperado = 2
        
        resultado = birthday(s, d, m)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        n = 6
        s = [1, 1, 1, 1, 1, 1]
        d = 3
        m = 2
        
        esperado = 0
        
        resultado = birthday(s, d, m)
        
        assert resultado == esperado

File: algorithms/implementation/the-birthday-bar/the-birthday-bar-javascript/js/test_birthday.js
import assert from 'assert';
import { birthday } from '../js/birthday.js';

describe('birthday', function() {

    it('o número de maneiras que a barra pode ser dividida é 0', function() {

        let s = [1, 1, 1, 1, 1, 1];
        let d = 3;
        let m = 2;

        let esperado = 0;

        let resultado = birthday(s, d, m);

        assert.strictEqual(resultado, esperado)
    });
});


describe('birthday', function() {

    it('o número de maneiras que a barra pode ser dividida é 2', function() {

        let s = [1, 2, 1, 3, 2];
        let d = 3;
        let m = 2;

        let esperado = 2;

        let resultado = birthday(s, d, m);

        assert.strictEqual(resultado, esperado)
    });
});

describe('birthday', function() {

    it('o número de maneiras que a barra pode ser dividida é 1', function() {

        let s = [4];
        let d = 4;
        let m = 1;

        let esperado = 1;

        let resultado = birthday(s, d, m);

        assert.strictEqual(resultado, esperado)
    });
});